react/src/utils/Env.js

```diff
@@ -0,0 +1,280 @@
+        const pathSansBasePath = currentPath.substring(basePathWithTrailingSlash.length);
+        if (pathSansBasePath.length > 0) {
+            const slash = pathSansBasePath.indexOf("/");
+            env = (slash >= 0) ? pathSansBasePath.substring(0, slash) : pathSansBasePath;
+        }
+    }
+    if (Type.IsObject(header)) {
+        if (!IsKnownEnv(env, header)) {
+            env = GetDefaultEnv(header);
+            // env = header?.default_env;
+        }
+    }
+    return env;
+}
+
+function IsCurrentEnv(env) {
+    return AreSameEnvs(GetCurrentEnv(), env);
+}
+
+function IsCurrentEnvKnown(header) {
+    return IsKnownEnv(GetCurrentEnv(), header);
+}
+
+function IsCurrentEnvAllowed(header) {
+    return IsAllowedEnv(GetCurrentEnv(), header);
+}
+
+// -------------------------------------------------------------------------------------------------
+// Environment name variants related functions.
+// -------------------------------------------------------------------------------------------------
+
+function GetAnnotatedEnv(env, header) {
+    if (Type.IsObject(env)) {
+        return env;
+    }
+    else if (Str.HasValue(env)) {
+        const knownEnvs = GetKnownEnvs(header);
+        for (const knownEnv of knownEnvs) {
+            if (AreSameEnvs(knownEnv, env)) {
+                return knownEnv;
+            }
+        }
+    }
+    return null;
+}
+
+function GetRegularEnvName(env, header) {
+    return GetAnnotatedEnv(env, header)?.name;
+}
+
+function GetPublicEnvName(env, header) {
+    return GetAnnotatedEnv(env, header)?.public_name;
+}
+
+function GetFullEnvName(env, header) {
+    return GetAnnotatedEnv(env, header)?.full_name;
+}
+
+function GetShortEnvName(env, header) {
+    return GetAnnotatedEnv(env, header)?.short_name;
+}
+
+function GetFoursightEnvName(env, header) {
+    return GetAnnotatedEnv(env, header)?.foursight_name;
+}
+
+function GetPreferredEnvName(env, header) {
+    return IsFoursightFourfront(header) ? GetPublicEnvName(env, header) : GetFullEnvName(env, header);
+}
+
+// -------------------------------------------------------------------------------------------------
+// Foursight-Fourfront vs. Foursight-CGAP and legacy Foursight related functions.
+// -------------------------------------------------------------------------------------------------
+
+function IsFoursightFourfront(header) {
+    if (Cookie.TestMode.HasFoursightFourfront()) {
+        return true;
+    }
+    else if (Cookie.TestMode.HasFoursightCgap()) {
+        return false;
+    }
+    else {
+        return header?.app?.package !== "foursight-cgap";
+    }
+}
+
+function GetLegacyFoursightLink(header) {
+    //
+    // For Foursight-CGAP (as opposed to Foursight-Fourfront) going to just,
+    // for example, /api/view/supertest, does not work, rather we want to
+    // to, for example, /api/view/cgap-supertest. I.e. for Foursight-CGAP
+    // use the full name and the public name for Foursight-Fourfront.
```

Comment:
This seems like a bug in the back-end? I think both the long and short name should work here?

Reply:
I think I might need help. At some point a little while back I think I came up with a fix in `check_utils.py` in `get_grouped_check_results` by using `connection.ff_env` rather than `connection.fs_env` but wasn't sure that was the right thing.